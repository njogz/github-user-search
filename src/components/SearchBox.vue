<template>
  <v-row>
      <v-col>
        <v-text-field
          v-model="searchQuery"
          label="Search users"
        ></v-text-field>
      </v-col>
      <v-col>
        <v-btn class="mr-4" @click="search">search</v-btn>
      </v-col>
  </v-row>
</template>
<script>
export default {
  data: () => ({
    searchQuery: '',
  }),
  methods: {
    search() {
      this.$store.dispatch('searchUsers', { query: this.searchQuery, page: 1 });
    },
  },
};
</script>
