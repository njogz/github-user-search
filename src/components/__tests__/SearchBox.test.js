import { shallowMount, createLocalVue, mount } from '@vue/test-utils';
import Vuetify from 'vuetify';
import Vuex from 'vuex';
import SearchBox from '@/components/SearchBox.vue';

const localVue = createLocalVue();
localVue.use(Vuetify);
localVue.use(Vuex);

describe('SearchBox.vue', () => {
  let vuetify;
  let actions;
  let store;

  beforeEach(() => {
    vuetify = new Vuetify();
    actions = {
      searchUsers: jest.fn(),
    };
    store = new Vuex.Store({
      actions,
    });
  });

  it('Should match the saved snapshot', () => {
    const wrapper = shallowMount(SearchBox, {
      localVue,
      vuetify,
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Should trigger a vuex action with whatever is in the text field', () => {
    const wrapper = mount(SearchBox, {
      localVue,
      vuetify,
      store,
    });
    wrapper.find('[data-query]').setValue('alice');
    wrapper.find('[data-search]').trigger('click');
    expect(actions.searchUsers.mock.calls[0][1]).toEqual({ page: 1, query: 'alice' });
  });
});
