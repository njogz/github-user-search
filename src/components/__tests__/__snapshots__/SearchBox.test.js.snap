// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox.vue Should match the saved snapshot 1`] = `
<v-row-stub tag="div">
  <v-col-stub tag="div">
    <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" value="" backgroundcolor="" label="Search users" loaderheight="2" clearicon="$clear" type="text" data-query=""></v-text-field-stub>
  </v-col-stub>
  <v-col-stub tag="div">
    <v-btn-stub tag="button" activeclass="" type="button" data-search="" class="mr-4">search</v-btn-stub>
  </v-col-stub>
</v-row-stub>
`;
